csrc/scheduler/normalization_utils.cpp

```diff
@@ -1123,6 +1123,24 @@ bool compileTimeCheck(Fusion* fusion, ScheduleHeuristic schedule_heuristic) {
   return true;
 }
 
+int64_t getSharedMemoryOverheadPerBlock(
+    Fusion* fusion,
+    const std::vector<TensorView*>& reduction_tvs,
+    const int64_t max_threads_per_block) {
+  const auto& dev_prop = at::cuda::getCurrentDeviceProperties();
+  int64_t dtype_size = 1;
+  for (auto tv : reduction_tvs) {
+    dtype_size = std::max(dtype_size, dataTypeSize(tv->getDataType().value()));
+  }
+  int64_t welford_factor = ir_utils::hasOpsOfType<WelfordOp>(fusion) ? 3l : 1l;
+  int64_t reduction_broadcast_workspace =
```

Comment:
Suppose we have a welford op with `float` and `int64`. Could you remind if we allocate `sizeof(int64_t) * num_threads`? Or is it `(sizeof(float) * 2 + sizeof(int64_t)) * num_threads`?

Reply:
It is actually
```
  nvfuser_index_t block_size = blockDim.x*blockDim.y*blockDim.z;
  nvfuser_index_t *shared_mem_var = static_cast<nvfuser_index_t*>(shared_mem);
  nvfuser_index_t *shared_mem_avg = shared_mem_var + block_size;
  nvfuser_index_t *shared_mem_n = shared_mem_avg + block_size;
```
I revised the code accordingly.